dashboard/app/models/pd/workshop.rb

```diff
@@ -213,8 +213,30 @@ def self.start_on_or_after(date)
 
   # Orders by the scheduled start date (date of the first session),
   # @param :desc [Boolean] optional - when true, sort descending
-  def self.order_by_start(desc: false)
-    joins(:sessions).group(:pd_workshop_id).order('DATE(MIN(start))' + (desc ? ' DESC' : ''))
+  def self.order_by_scheduled_start(desc: false)
+    joins(:sessions).
+    group('pd_workshops.id').
+    order('DATE(MIN(pd_sessions.start))' + (desc ? ' DESC' : ''))
+  end
+
+  # Orders by the number of active enrollments
+  # @param :desc [Boolean] optional - when true, sort descending
+  def self.order_by_enrollment_count(desc: false)
+    left_outer_joins(:enrollments).
+    group('pd_workshops.id').
+    order('COUNT(pd_enrollments.id)' + (desc ? ' DESC' : ''))
+  end
+
+  # Orders by the workshop state, in order: Not Started, In Progress, Ended
+  # @param :desc [Boolean] optional - when true, sort descending
+  def self.order_by_state(desc: false)
```

Comment:
I'm a little twitchy at all of this raw SQL - I feel like there is a way to write this in ActiveRecord